packages/valory/skills/hello_world_abci/rounds.py

```diff
@@ -152,14 +162,34 @@ class ResetAndPauseRound(CollectSameUntilThresholdRound, HelloWorldABCIAbstractR
     def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
         """Process the end of the block."""
         if self.threshold_reached:
-            return self.synchronized_data.create(), Event.DONE
+            # TODO: Question: Seems like we can just not re-create the synchronized data here, to keep the print_count alive. But is that the right approach?
+            return self.synchronized_data, Event.DONE
         if not self.is_majority_possible(
             self.collection, self.synchronized_data.nb_participants
         ):
             return self.synchronized_data, Event.NO_MAJORITY
         return None
 
 
+class PrintMessageCountRound(
+    CollectSameUntilThresholdRound, HelloWorldABCIAbstractRound
+):
+    payload_class = PrintMessageCountPayload
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        if self.threshold_reached:
+            # TODO: Question: Why not have most voted for the whole `PrintMessageCountPayload`? But instead it's per-value?
```

Comment:
In principle, the parent classes agree on all params of the payload, except "sender", "round_count", "id_". The user then decides which of these parameters of the payload need to be transferred to the syncrhonized data.